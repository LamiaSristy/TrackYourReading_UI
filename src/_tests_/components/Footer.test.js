import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Footer from '../../components/Footer';

const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <Footer />
  </Provider>,
);

describe('Footer Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  test('renders the menu the page', () => {
    const page = component.find('.menu');
    expect(page).toMatchSnapshot();
  });

  test('renders the text of the icons class', () => {
    const text = component.find('.icons');
    expect(text).toMatchSnapshot();
  });

  test('renders the text of the icon-btn class', () => {
    const text = component.find('.icon-btn');
    expect(text).toMatchSnapshot();
  });

  test('renders the text of the add-trackings class', () => {
    const text = component.find('.add-trackings');
    expect(text).toMatchSnapshot();
  });
});
