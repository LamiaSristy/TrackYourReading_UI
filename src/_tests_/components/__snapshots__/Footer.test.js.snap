// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders the menu the page 1`] = `ShallowWrapper {}`;

exports[`Footer Component renders the text of the add-trackings class 1`] = `ShallowWrapper {}`;

exports[`Footer Component renders the text of the icon-btn class 1`] = `ShallowWrapper {}`;

exports[`Footer Component renders the text of the icons class 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the form the page 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the menu the page 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the text of the add-trackings class 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the text of the buttons-form class 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the text of the icon-btn class 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the text of the icons class 1`] = `ShallowWrapper {}`;

exports[`Testpage Component renders the text of the one-parameter class 1`] = `ShallowWrapper {}`;
